quintilesims/layer0
# 553: Support ELB Idle Timeout in CLI and Terraform when creating or updating load balancer

**What does this pull request do?**
Adds CLI functionality for `l0 loadbalancer create` and a new subcommand `l0 loadbalancer idletimeout`. Create has a new optional flag `--idle-timeout` for create which can be used to specify the Idle Timeout of the load balancer at creation time. `l0 loadbalancer idletimeout` is an update subcommand that updates a given load balancer's idle timeout to a new value.

Adds Terraform support for defining a load balancer's Idle Timeout. A new key is introduced into the resource, `idle_timeout`.

**How should this be tested?**
Manually via the CLI: `l0 loadbalancer create env_a lb_a --idle-timeout 100` and `l0 loadbalancer idletimeout lb_a 200`

Via terraform:
```
provider "layer0" {
  endpoint        = "${var.endpoint}"
  token           = "${var.token}"
  skip_ssl_verify = true
}

resource "layer0_environment" "env_a" {
  name = "env_a"
}

resource "layer0_load_balancer" "lb_a" {
  name        = "lb_a"
  environment = "${layer0_environment.env_a.id}"

  port {
    host_port      = 80
    container_port = 80
    protocol       = "http"
  }

  idle_timeout = 75
}
```
Apply, then change idle_timeout and apply again.

Unit tests have been created and they should pass.  Documentation has been updated to reflect these changes.

**Checklist**
~- [ ] Smoke tests (if applicable)~
~- [ ] System tests (if applicable)~

closes #553
